tests/src/sanity.rs

```diff
@@ -0,0 +1,284 @@
+        let pod_namespace = "default";
+        let env: HashMap<&str, &str> = HashMap::new(); // for adding more environment variables
+        let mut server = test_server_init(&client, pod_namespace, env).await;
+
+        let service_url = get_service_url(&client, pod_namespace).await.unwrap();
+
+        let mut stderr_reader = BufReader::new(server.stderr.take().unwrap());
+        let child_stdout = server.stdout.take().unwrap();
+
+        tokio::spawn(async move {
+            loop {
+                let mut error_stream = String::new();
+                // the following is a blocking call, so we need to spawn a task to read the stderr
+                stderr_reader.read_line(&mut error_stream).await.unwrap();
+                // Todo(investigate): it looks like the task reaches the panic when not kept in a
+                // loop i.e. the buffer reads an empty string:  `thread 'main'
+                // panicked at 'Error: '`
+                if !error_stream.is_empty() {
+                    panic!("Error: {}", error_stream);
+                }
+            }
+        });
+
+        // since we are reading from the stdout, we could block at any point in case the server
+        // does not write to its stdout, so we need a timeout here
+        let validation_timeout = Duration::from_secs(20);
+        timeout(
+            validation_timeout,
+            validate_requests(child_stdout, service_url.as_str()),
+        )
+        .await
+        .unwrap();
+        server.kill().await.unwrap();
+        // look for the agent job and pod
+        let jobs_api: Api<Job> = Api::namespaced(client.clone(), "default");
+        let jobs = jobs_api.list(&ListParams::default()).await.unwrap();
+        // assuming only one job is running
+        // to make the tests parallel we need to figure a way to get the exact job name when len() >
+        // 1
+        assert_eq!(jobs.items.len(), 1);
+
+        let pods_api: Api<Pod> = Api::namespaced(client.clone(), "default");
+        let pods = pods_api.list(&ListParams::default()).await.unwrap();
+        assert_eq!(pods.items.len(), 2);
+
+        let cleanup_timeout = Duration::from_secs(35);
+        timeout(
+            cleanup_timeout,
+            tokio::spawn(async move {
+                // verify cleanup
+                loop {
+                    let updated_jobs = jobs_api.list(&ListParams::default()).await.unwrap();
+                    let updated_pods = pods_api.list(&ListParams::default()).await.unwrap(); // only the nginx pod should exist
+                    if updated_pods.items.len() == 1 && updated_jobs.items.len() == 0 {
+                        let nginx_pod = updated_pods.items[0].metadata.name.clone().unwrap();
+                        assert!(nginx_pod.contains("nginx"));
+                        break;
+                    }
+                }
+            }),
+        )
+        .await
+        .unwrap()
+        .unwrap();
```

Comment:
@aviramha @meowjesty do you guys think this timeout on job cleanup is okay? or is there a better way to do it?